docs/data/date-pickers/adapters/adapters.md

```diff
@@ -0,0 +1,93 @@
+---
+title: Library to manipulate dates
+---
+
+# Library to manipulate dates
+
+<p class="description">Choose which library the pickers must use to manipulate the dates.</p>
+
+## Why do you need a library ?
+
+Like most pickers, the MUI Date and Time Pickers need a library to manipulate the dates.
+It will be used to format, parse and mutate the date inside all of our components.
+
+The specificity of the MUI Date and Time Pickers is to let you choose which library you want to use for the date manipulation.
+This lets you pick the library you already use in your application, without adding an extra one in your bundle.
+
+To achieve this, both `@mui/x-date-pickers` and `@mui/x-date-pickers-pro` exports a set of **adapters** which exposes
+the date manipulation libraries under a unified api.
+
+## Choosing a date library
+
+### Available libraries
+
+The Date and Time Pickers currently support the following date libraries:
+
+- [Day.js](https://day.js.org/)
+- [date-fns](https://date-fns.org/)
+- [Luxon](https://moment.github.io/luxon/#/)
+- [Moment.js](https://momentjs.com/)
+
+:::info
+If you are using another calendar system than the _Gregorian_ one (i.e: _Jalali_ or _Hijri_ calendars),
+please refer to the [Support for other calendar systems](#support-for-other-calendar-systems) section.
+:::
+
+### Recommended library
+
+If you are already using one of these libraries in your application, you can keep using it for the Date and Time Pickers as well.
+This will avoid bundling two libraries.
+
+If you are starting a new project without any date manipulation outside of `@mui/x-date-pickers`,
+consider using `dayjs` which will have the smallest impact on the bundle size of your application.
+
+Here is the weight added to your gzipped bundle size by each of those libraries when used inside the Date and Time Pickers:
+
+| **Library**       | **Gzipped size** |
+| ----------------- | ---------------- |
+| `dayjs@1.11.5`    | 6.77kb           |
+| `date-fns@2.29.3` | 19.39kb          |
+| `luxon@3.0.4`     | 23.26kb          |
+| `moment@2.29.4`   | 20.78kb          |
```

Comment:
> Have you considered being a bit less opinionated about the build process and not rely on specific versions for this documentation? thinking


How would you compute those numbers ? 
I can't rely on the ones advertised by the libraries because 
- we can tree shake some, not others
- for dayjs we uses plugins
- it would not take the `Adapter` itself into account

If my results are correct, `luxon` is ofr example a lot worse than expected.
And for me it's because they use very short variables name in their code, so the minification process does not have a lot to do.

Reply:
Yeah, those are all very valid points. I do agree that realistically your numbers do give a lot more realistic bundle size diff representation. 👌 
I was just asking if you considered a less accurate, but lazy (no need to update to reflect changes) approach? 🙈 😆

Reply:
I feel like the advertised library size is too far from the reality and that the data I computed will still be more relevant even in 2026 :laughing: 

But I can understand that it's not super appealing

Reply:
I tend to agree that your provided numbers will be way more relevant to most users. We can safely close this discussion. 👌